lib/MemoryModel/LocMemModel.cpp

```diff
@@ -60,7 +60,7 @@ bool LocSymTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         if (index <= baseIndex) {
             /// variant offset
             // Handling pointer types
-            if (const PointerType* pty = dyn_cast<PointerType>(*gi)) {
+	  if (const PointerType* pty = dyn_cast<PointerType>(gi.getIndexedType())) {
```

Comment:
For LocMemModel, we can also get rid of the `gep_type_iterator` and `gi.getIndexedType()`, and stick with your BridgedGepIterator for now:).

Please use `*gi` intead of `gi.getIndexedType()` in LocMemModel. Please also replace `gep_type_iterator` at lines 50 and 57 with the BridgedGepIterator.